src/plugins/administrative_templates/ui/presentationbuilder.cpp

```diff
@@ -107,6 +107,107 @@ QHBoxLayout *createCaptions()
     return horizontalLayout;
 }
 
+QMap<std::string, QString> loadListFromRegistry(AbstractRegistrySource &m_source, const std::string &key, const std::string &prefix)
+{
+    QMap<std::string, QString> items;
+    std::vector<std::string> valueNames = m_source.getValueNames(key);
+    
+    // finding and erase **delvals.
+    for (auto it = valueNames.begin(); it != valueNames.end(); ++it) {
+        if (QString::fromStdString(*it).compare("**delvals.", Qt::CaseInsensitive) == 0) {
+            it = valueNames.erase(it);
+
+            // std::vecotor<T>::erase() returning iterator that pointing 
+            // to the next element after erased
+            --it; 
+            break;
+        }
+    }
```

Comment:
```c++
valueNames.erase(std::remove_if(valueNames.begin(), valueNames.end(),
                                [](const std::string& name) {
                                    return name.find("**delvals.", 0) == 0;
                                }),
                 valueNames.end());
```


- std::remove_if is used to remove elements from the range [valueNames.begin(), valueNames.end()) for which the predicate (lambda function) returns true.
- The lambda function checks if the string starts with "**delvals." by using the find method, which returns the position of the first occurrence of the substring. If the substring is found at the beginning of the string (i.e., find returns 0), the predicate returns true.
- The resulting range of elements to be removed is then passed to erase, which actually removes the elements from the vector.

Reply:
If we absolutely want to use case-insensitive comparison we could probably use `strnicmp` function like this:
```c++
strnicmp(str.c_str(), substr.c_str(), substr.size());
```
From manual:
RETURN VALUES
The `strcmp()` and `strnicmp()` functions return an integer greater than, equal to, or less than 0, according as the string s1 is greater than, equal to, or less than the string s2.  The comparison is done using unsigned characters, so that '200' is greater than ''.

Reply:
Thanks! 
`strincmp` is missing from 'cstring'. For now, I will use `QString::compare`.